lldb/source/Expression/DWARFExpression.cpp

```diff
@@ -2015,8 +2015,6 @@ llvm::Expected<Value> DWARFExpression::Evaluate(
       if (stack.size() < 1) {
         UpdateValueTypeFromLocationDescription(log, dwarf_cu,
                                                LocationDescriptionKind::Empty);
-        // Reset for the next piece.
-        dwarf4_location_description_kind = Memory;
```

Comment:
I'm a bit surprised by these ones. Why aren't these needed anymore?

Reply:
It's a local variable in the `DW_OP_piece` block, and we return with an error here.  It's a dead store.

Reply:
It might be that stylistically we want to set `dwarf4_location_description_kind` in every clause of this DW_OP_piece block, and we should set it to stay consistent.